### PR TITLE
[docs] Added description of saving using ddp

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Clears up that lightning handles ddp saving using our in built functions.

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/weights_loading.rst [hunk 1]
```diff
@@ -14,6 +14,8 @@ Lightning automates saving and loading checkpoints. Checkpoints capture the exac
 
 Checkpointing your training allows you to resume a training process in case it was interrupted, fine-tune a model or use a pre-trained model for inference without having to retrain the model.
 
+Lightning also takes care of any accelerators including ddp, where multiple processes are used. In this scenario, we only save on the main process.
```
Comment (edenlightning) [inline docs/source/weights_loading.rst:17]: This could use some clarification. How does saving DDP differ from regular models?

Can we put an example?

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: docs/source/weights_loading.rst (hunk 1 above)
Comment (SeanNaren) [reply docs/source/weights_loading.rst]: @edenafek thanks for the comment! I've updated the branch with an example in the docs, let me know if this is closer to what you were after :)

File: docs/source/weights_loading.rst [hunk 2]
```diff
@@ -137,6 +138,19 @@ You can manually save checkpoints and restore your model from the checkpointed s
     trainer.save_checkpoint("example.ckpt")
     new_model = MyModel.load_from_checkpoint(checkpoint_path="example.ckpt")
 
+Manual saving with DDP
+======================
+
+Lightning also handles accelerators such as DDP, where multiple processes are used. In this scenario, we only save on the main process.
```
Comment (edenlightning) [inline docs/source/weights_loading.rst:144]: Can you elaborate? What does this mean? Can you resume training using a single process? only multi process? Does that change anything about what is being saved?

File: docs/source/weights_loading.rst [hunk 3]
```diff
@@ -137,6 +138,20 @@ You can manually save checkpoints and restore your model from the checkpointed s
     trainer.save_checkpoint("example.ckpt")
     new_model = MyModel.load_from_checkpoint(checkpoint_path="example.ckpt")
 
+Manual saving with accelerators
+======================
+
+Lightning also handles accelerators such as DDP where multiple processes are required. For example, using DDP our training script is run across multiple devices automatically to parallelize training.
```
Comment (edenlightning) [inline docs/source/weights_loading.rst:144]: when using DDP...

File: docs/source/weights_loading.rst [hunk 4]
```diff
@@ -137,6 +138,20 @@ You can manually save checkpoints and restore your model from the checkpointed s
     trainer.save_checkpoint("example.ckpt")
     new_model = MyModel.load_from_checkpoint(checkpoint_path="example.ckpt")
 
+Manual saving with accelerators
+======================
```
Comment (rohitgr7) [inline docs/source/weights_loading.rst:142]:
```suggestion
Manual saving with accelerators
===============================
```

File: docs/source/weights_loading.rst (hunk 3 above)
Comment (rohitgr7) [inline docs/source/weights_loading.rst:144]:
```suggestion
Lightning also handles accelerators such as DDP where multiple processes are required. For example, when using DDP our training script is run across multiple devices automatically to parallelize training.
```

**Review by rohitgr7 — APPROVED** [review]

**Comment by SeanNaren** [comment]
Do not merge, still feedback to address

**Review by awaelchli — APPROVED** [review]

File: docs/source/weights_loading.rst [hunk 5]
```diff
@@ -137,6 +138,20 @@ You can manually save checkpoints and restore your model from the checkpointed s
     trainer.save_checkpoint("example.ckpt")
     new_model = MyModel.load_from_checkpoint(checkpoint_path="example.ckpt")
 
+Manual saving with accelerators
```
Comment (edenlightning) [inline docs/source/weights_loading.rst:141]: Perhaps it will be better to just add a warning that lightning makes sure saving is done only on one process when running using accelerators, but the behaviour is not guaranteed when using loggers that don't use lightning default save, such as mlflow.

Comment (SeanNaren) [reply docs/source/weights_loading.rst:141]: I added a description of how to remedy the situation if you use custom loading functions. In the case of mlflow, I think this would be enough for the team to resolve their issue with saving!

Adding a warning would be nice, but a bit tricky since we would need to catch how mlflow saves the model, and override pytorch state_dicts which would be messy. I'll sync up with the mlflow team, but hopefully the additional doc change I added explains verbose enough for them

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source/weights_loading.rst [hunk 6]
```diff
@@ -137,6 +138,23 @@ You can manually save checkpoints and restore your model from the checkpointed s
...
+
+.. code-block:: python
+
+    trainer = Trainer(accelerator="ddp")
+    model = MyLightningModule(hparams)
+    trainer.fit(model)
+    # Saves only on the main process
+    trainer.save_checkpoint("example.ckpt")
+
+Not using `trainer.save_checkpoint` can lead to unexpected behaviour and potential deadlock. Using other saving functions will result in all devices attempting to save the checkpoint. As a result, we highly recommend using the trainer's save functionality.
```
Comment (edenlightning) [inline docs/source/weights_loading.rst:155]: Consider adding "(for example, some loggers such as MLFlow do not use lightning built-in save function)
